lib/src/command/lish.dart

```diff
@@ -154,9 +161,13 @@ the \$PUB_HOSTED_URL environment variable.''',
     if (!isValid) {
       overrideExitCode(exit_codes.DATA);
       return;
-    } else if (dryRun) {
-      return;
-    } else {
+    }
+
+    if (saveTar != null) {
+      await File(saveTar).writeAsBytes(await packageBytesFuture);
```

Comment:
If we go through with this we need a `writeBinaryFile` in `lib/src/io.dart` that logs how many bytes are written to which location (should not write the contents) analogous  to `writeTextFile`.